src/network/providers/solana.ts

```diff
@@ -30,12 +34,40 @@ interface TransactionInstructionJSON {
     data: number[];
 }
 
-const createSolanaConnection = (blockchainUrl: string) => {
-    const connection = new web3.Connection(
-        blockchainUrl,
-        'confirmed'
-    );
+enum Protocol {
+    IPFS = 'ipfs',
+    ARWV = 'arwv'
+}
+
+type DomainContent = {
+    protocolType: Protocol | null;
+    decoded: string | null;
+    error?: Error;
+};
+
+type DomainRegistry = {
```

Comment:
`web3.eth.ens` and `@solana/web3.js` return data in different formats. The goal here is to standardize the return value of the `resolveDomain` methods, so that, ultimately, the API always return the same response format, regardless of the blockchain.

This should also be clearer and easier to enforce when we add an abstraction for the providers.